src/sentry/migrations/0384_backfill_monitor_env_checkins.py

```diff
@@ -0,0 +1,82 @@
+# Generated by Django 2.2.28 on 2023-03-14 23:45
+
+from django.db import connection, migrations
+from psycopg2.extras import execute_values
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+BATCH_SIZE = 100
+
+UPDATE_QUERY = """
+    UPDATE sentry_monitorcheckin
+    SET monitor_environment_id = data.monitor_environment_id
+    FROM (VALUES %s) AS data (id, monitor_environment_id)
+    WHERE sentry_monitorcheckin.id = data.id"""
+
+
+def backfill_monitor_checkins(apps, schema_editor):
+    MonitorCheckIn = apps.get_model("sentry", "MonitorCheckIn")
+    MonitorEnvironment = apps.get_model("sentry", "MonitorEnvironment")
+
+    monitor_mappings = {
+        monitor_id: monitor_env_id
+        for monitor_id, monitor_env_id in MonitorEnvironment.objects.values_list("monitor_id", "id")
+    }
+
+    queryset = RangeQuerySetWrapperWithProgressBar(
+        MonitorCheckIn.objects.all().values_list(
+            "id",
+            "monitor_id",
+            "monitor_environment_id",
+        ),
```

Comment:
I want to make sure that:
1. we only return monitorenvs with the `env` named `production`
2. in the case of multiple, return the most recent.

I'm not sure we can do 1 without selecting `environment` objects as well. 2 we can order by `date_added`

Reply:
there's 2 instances of `1` in the db from one company

there's 1 instance of `2` from a monitor that moved projects

Reply:
For 1. you can do `filter(environment__name="production")` when you build `monitor_mappings`. Ordering by `date_added` for 2 makes sense to me

Reply:
Yeah I had 1 but my tests were failing. Realize now it was bad tests lol